enhancements/monitoring/monitoring-config.md

```diff
@@ -0,0 +1,190 @@
+
+## Summary
+
+* Currently, the monitoring stack is configured using a configmaps. In OpenShift though the best practice is to configure operators using custom resources.
+
+
+## Motivation
+
+* The specification is well known and to a degree self-documenting
+* The APIServer will validate user resources based on our specifications, so users get immediate feedback on errors instead of having to check if their config was applied and check logs.
+* Many users expect to interact with operators through a CRD
+* Compatible with GitOps workflows. 
+* We can add [cross]validation rules to CRD fields to avoid misconfigurations
+* End users get a much faster feedback loop. No more applying the config and scanning logs if things don't look right. The API server will give immediate feedback
+* Organizational users (such as ACM) can manage a single resource and observe its status
+
+### Goals
+
+- Replace configmaps with CRD
+- Smooth transition for users
+
+## Proposal
+
+### Overview
+
+- Replace confimgaps with CRD:
+  cluster-monitoring-configmap
+  user-workload-monitoring-config
+
+  pkg/manifests/config.go
+
+```
+  type Config struct {
+	Images                               *Images `json:"-"`
+	RemoteWrite                          bool    `json:"-"`
+	CollectionProfilesFeatureGateEnabled bool    `json:"-"`
+
+	ClusterMonitoringConfiguration *ClusterMonitoringConfiguration `json:"-"`
+	UserWorkloadConfiguration      *UserWorkloadConfiguration      `json:"-"`
+}
+```
+
+### Migration path
+
+- Feature gate. 
+- Move each component to openshift/api/config
+- Create two CRD's:
+  * ClusterMonitoringConfiguration
+  * UserWorkloadConfiguration
+  The reason for having two different CRDs is to address the needs of two distinct roles who manage them. In managed OpenShift clusters (OSD/ROSA), the platform SREs are responsible for the cluster (platform) monitoring stack, while the platform users (e.g., customers) handle the user-defined monitoring stack.º
+- Review/improve/modify every configmap component to create a compliant openshif/api type. Fix erros, clean old things, improve names and config.
+- Switch mecanishm
+- Both ConfigMap and CRD will coexist for a while until the transition is complete.
+- CRD and ConfigMap will be merged into a single structure, this structure will config CMO
+- Extensive testing will be required to validate the new behavior.
+- If there are different fields in CRD and ConfigMap, the field in the ConfigMap will take precedence.
+- Some fields in CRD and ConfigMap diverge due to API requirements. Helper functions will be needed to resolve this.
+- The end user will be able to choose to use ConfigMap or CRD at any time, keeping in mind that the fields in the CRD will gradually expand.
+
+
+### Issues
+
+- Merge CRD and configmap could be error prone and will need extra test
+- Change of types and names and how handle it when there are CRD and configmap
+
```

Comment:
should these items go to "Open Questions" for now?

Reply:
done